### DISCUSSION TITLE
Overwrite the tooltip label

### DISCUSSION THREAD

**Discussion by darthzeran** [TanStack/react-charts#107] [General] [answered]
Is there a way to change the label for the tooltip?
For example, my x axis is dates (Specifically, Date objects),
and I would like to overwrite how the tooltip shows my data points.  All it is doing is grabbing the getTime() of the date object, so the tooltip shows my data labelled as `561`, `334`, etc, representing the milliseconds of my Date object.

In the docs, I cannot find a way of changing that without creating a brand new tooltip. Any suggestions?

**Comment by evenstephenr** [comment]
@darthzeran Overriding the custom Tooltip is the recommended way to go, here's a codesandbox showing how to override the default Tooltips with a custom component.

https://codesandbox.io/s/github/tannerlinsley/react-charts/tree/next/examples/custom-tooltip

Does this help?

**Reply by darthzeran** [reply]
It throws an error, so unfortunately it does not, but I am still looking at the code to trial and error my way to progress.

**Comment by tannerlinsley** [comment, marked as answer]
Due to a massive overhaul and rewrite of React Charts to TypeScript and a massively upgraded public API, this discussion is being marked as answered for triage. The latest version can be tested via the `react-charts@beta` tag. Documentation is in the process of being upgraded, but thankfully, everything is fully typed now, so you can explore the API that way for now.

If, *after testing or playing with the v3 beta*, you believe this discussion is still relevant or unanswered, feel free to mark it as such and continue the discussion.
